### DISCUSSION TITLE
Update documentation of `shouldDirty` option in `setValue` for clarity.

### DISCUSSION THREAD

**Discussion by jviall** [react-hook-form/react-hook-form#6628] [Ideas] [answered]
I've been really confused as to how the `shouldDirty` option is made to work when using `setValue`. I've been wanting to utilize the api to "reset" an individual form field and clear its dirty state. Here's a codesandbox that achieves what I've been searching for: https://codesandbox.io/s/react-hook-form-formstate-dirty-touched-submitted-forked-opflm?file=/src/index.js

The documentation of `shouldDirty` defines it as a boolean prop that determines:
> Whether to set the input itself to dirty.

The way I understand it, it sounds like the value of `shouldDirty` that I pass will be reflected onto the `dirtyFields` state for all the fields that I am setting a value on. But really it's more like `shouldDirty` determines whether or not the "dirtiness" of the field(s) will be re-computed. So If I want to "reset" a text input (e.g. named `'text'` whose default value is an empty string, then the proper call is `setValue('text', '', { shouldDirty: true })`, and this will set value back to the default and remove the field from the `dirtyFields` state.

I'm not sure what the right wording is, but I found the current language to be very confusing.

**Comment by bluebill1049** [comment]
thanks for the feedback. would you like to send us a PR to improve on that.

**Reply by jviall** [reply]
I sure could! Is my understanding of the behavior correct then?

**Comment by bluebill1049** [comment, marked as answer]
what the API does:

shouldDirty:

- compute if your input should be dirty against your defaultValues (subscribed to `dirtyFields`)
- compute if your form is dirty or not against your defaultValues (subscribed to `isDrirty`)

shouldValidate:

- compute if your input is validated or not (subscribed to `errors`)
- compute if your entire form is valid or not (subscribed to `isValid`)

**Reply by jviall** [reply]
Thanks!
